### PR TITLE
SeekableIterator extends Iterator 

### PR DESCRIPTION
**Pull request by WinterSilence**
http://docs.php.net/manual/en/class.seekableiterator.php

### CODE REVIEW

File: classes/Kohana/Database/Result.php [hunk 1]
```diff
@@ -8,7 +8,7 @@
  * @copyright  (c) 2008-2009 Kohana Team
  * @license    http://kohanaphp.com/license
  */
-abstract class Kohana_Database_Result implements Countable, Iterator, SeekableIterator, ArrayAccess {
+abstract class Kohana_Database_Result implements ArrayAccess, SeekableIterator, Countable {
```
Comment (shadowhand) [inline classes/Kohana/Database/Result.php:11]: why has the order been changed?

Comment (WinterSilence) [inline classes/Kohana/Database/Result.php:11]: @shadowhand interfaces are ordered by importance

Comment (kemo) [inline classes/Kohana/Database/Result.php:11]: how do you calculate 'importance'?

Comment (WinterSilence) [inline classes/Kohana/Database/Result.php:11]: @kemo you have classes:

```
class A implements ArrayAccess, SeekableIterator {}
class B implements ArrayAccess, Countable {}
class C implements SeekableIterator, Countable {}
```

Which of the classes is most similar to an array?

But main thing is remove duplication interface.

**Comment by mintbridge** [comment]
I know we don't follow PSR standards, and even then its not stated explicitly but from the examples its looks like they list them alphabetically. https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md#41-extends-and-implements

**Comment by evanpurkhiser** [comment]
Alphabetical ordering makes sense to me. I don't understand how you determine 'importance' of each interface, even with @WinterSilence's example :confused:

**Comment by WinterSilence** [comment]
@EvanPurkhiser @mintbridge interface ArrayAccess is mandatory, so he was the first, Cycles (foreach) is more uses than count, so that the second is SeekableIterator. In this case the order is not important, just in their classes, I always located interfaces in order of importance.

**Comment by mintbridge** [comment]
I agree with @EvanPurkhiser and @kemo , I don't think there is an order of importance for interfaces, and using implements means that they are mandatory by design. I agree its not important but I do think alphabetical makes more sense, so if you re-order the names to be alphabetical then this can be merged.

**Pull request closed without merging**